src/pages/workspace/companyCards/addNew/CardTypeStep.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {useOnyx} from 'react-native-onyx';
+import type {ValueOf} from 'type-fest';
+import Button from '@components/Button';
+import HeaderWithBackButton from '@components/HeaderWithBackButton';
+import Icon from '@components/Icon';
+import * as Illustrations from '@components/Icon/Illustrations';
+import ScreenWrapper from '@components/ScreenWrapper';
+import SelectionList from '@components/SelectionList';
+import RadioListItem from '@components/SelectionList/RadioListItem';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import * as CompanyCards from '@userActions/CompanyCards';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+
+function CardTypeStep() {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const [addNewCard] = useOnyx(ONYXKEYS.ADD_NEW_COMPANY_CARD);
+    const [typeSelected, setTypeSelected] = useState<ValueOf<typeof CONST.COMPANY_CARDS.CARD_TYPE>>();
+
+    const submit = () => {
+        CompanyCards.setAddNewCompanyCardStepAndData({
+            step: CONST.COMPANY_CARDS.STEP.CARD_INSTRUCTIONS,
+            data: {
+                cardType: typeSelected,
+            },
+            isEditing: false,
+        });
+    };
+
+    useEffect(() => {
+        setTypeSelected(addNewCard?.data.cardType);
+    }, [addNewCard?.data.cardType]);
+
+    const handleBackButtonPress = () => {
+        CompanyCards.setAddNewCompanyCardStepAndData({step: CONST.COMPANY_CARDS.STEP.CARD_TYPE});
```

Comment:
I think CardType back should close the RHP

https://github.com/user-attachments/assets/ebb3c775-0120-4b91-b6db-b6b9a2cc8b2c

Reply:
Makes sense! Will update soon